astropy/coordinates/sky_coordinate.py

```diff
@@ -532,6 +532,43 @@ def insert(self, obj, values, axis=0):
 
         return out
 
+    def is_transformable_to(self, new_frame):
+        """
+        Determines if this coordinate frame can be transformed to another
+        given frame.
+
+        Parameters
+        ----------
+        new_frame : frame class, frame object, or str
+            The proposed frame to transform into.
+
+        Returns
+        -------
+        transformable : bool or str
+            `True` if this can be transformed to ``new_frame``, `False` if
+            not, or the string 'same' if ``new_frame`` is the same system as
+            this object but no transformation is defined.
+
+        Notes
+        -----
+        A return value of 'same' means the transformation will work, but it will
+        just give back a copy of this object.  The intended usage is::
+
+            if coord.is_transformable_to(some_unknown_frame):
+                coord2 = coord.transform_to(some_unknown_frame)
+
+        This will work even if ``some_unknown_frame``  turns out to be the same
+        frame class as ``coord``.  This is intended for cases where the frame
+        is the same regardless of the frame attributes (e.g. ICRS), but be
+        aware that it *might* also indicate that someone forgot to define the
+        transformation between two objects of the same frame class but with
+        different attributes.
+        """
+        # TODO! like matplotlib, do string overrides for modified methods
+        new_frame = (_get_frame_class(new_frame) if isinstance(new_frame, str)
```

Comment:
No need for the `if` here - `_get_frame_class` already does that.

Reply:
``_get_frame_class`` errors on frame instances and ``frame.is_transformable_to`` errors on strings, so I need to convert string inputs (and not frame instances) before passing to ``frame.is_transformable_to``. Perhaps a more intuitive check would be 

``new_frame = (_get_frame_class(new_frame) if not isinstance(new_frame, BaseCoordinateFrame) else new_frame)``

It's functionally equivalent, but unnecessarily passes frame classes through ``_get_frame_class`` when it will be processed again in ``frame.is_transformable_to``.

Reply:
Oops, I missed that. Just let things be as they stand, then!